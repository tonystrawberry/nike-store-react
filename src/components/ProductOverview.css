.product-overview{
  max-width: 592px;
  margin: 10px;
  cursor: pointer;
}

.product-photo__container{
  width: 256px;
  height: 256px;
  overflow: hidden;
}

.product-photo{
  width: 256px;
}

.product-overview__info{
  padding-top: 10px;
  padding-left: 5px;
}

.product-overview__title{
  margin: 0;
  padding: 0;
  font-weight: 300;
  font-size: 16px;
}

.product-overview__type{
  color: #acacac;
}