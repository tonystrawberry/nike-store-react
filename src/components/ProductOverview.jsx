import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { selectProduct } from '../redux/actions';

import NikeAirForceSmall from '../assets/nike_airforce-small.jpg';

import './ProductOverview.css';

const mapStateToProps = (state) => {
  return {

  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onSelected: (productId) => dispatch(selectProduct(productId))
  }
}

class ProductOverview extends PureComponent {
  render() {
    return (
      <div className="product-overview" onClick={() => this.props.onSelected(1)}>
        <div className="product-photo__container">
          <img className="product-photo" src={NikeAirForceSmall} srcSet={`${NikeAirForceSmall} 216w`} alt="Nike Air Force"></img>
        </div>
        <div className="product-overview__info">
          <h1 className="product-overview__title">Nike Air Force 1</h1>
          <div className="product-overview__type">Men's Shoe</div>
          <div className="product-overview__type">2 Colors</div>
        </div>
      </div>
    );
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(ProductOverview);