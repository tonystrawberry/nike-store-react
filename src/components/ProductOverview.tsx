import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { selectProduct } from '../redux/actions';
import { Dispatch } from 'redux';

import './ProductOverview.scss';
import { Product } from '../types';

const mapStateToProps = () => {
  return {

  }
}

interface IProductOverviewProps {
  onSelected: {( productId : string): void},
  product: Product
}

interface IProductOverviewState {
  
}

const mapDispatchToProps = (dispatch : Dispatch) => {
  return {
    onSelected: (productId : string) => dispatch(selectProduct(productId))
  }
}

class ProductOverview extends PureComponent<IProductOverviewProps, IProductOverviewState> {
  render() {
    return (
      <div className="product-overview" onClick={() => this.props.onSelected(this.props.product._id)}>
        <div className="product-photo__container">
          <img className="product-photo" src={this.props.product.imageUrl} alt="Nike Air Force"></img>
        </div>
        <div className="product-overview__info">
          <h1 className="product-overview__title">{this.props.product.title}</h1>
          <div className="product-overview__type">{this.props.product.subtitle1}</div>
          <div className="product-overview__type">{this.props.product.subtitle2}</div>
        </div>
      </div>
    );
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(ProductOverview);