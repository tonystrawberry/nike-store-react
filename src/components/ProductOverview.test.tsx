import { mount, shallow } from 'enzyme';
import ProductOverview from './ProductOverview';
import React, { Suspense } from 'react';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';


const mockStore = configureStore([]);

it ('expect to render ProductOverview component', () => {
  let store = mockStore({});
  let wrapper = mount(<Provider store={store}><ProductOverview /></Provider>);

  expect(toJson(wrapper)).toMatchSnapshot();
})

it ('dispatch on .product-overview click', () => {
  let store = mockStore({});

  let wrapper = mount(
    <Provider store={store}>
      <ProductOverview />
    </Provider>
  );
  let productOverview = wrapper.find('.product-overview');
  productOverview.simulate('click');
  expect(store.getActions().length).toBe(1);
  expect(store.getActions()).toEqual([{type: "SELECT_PRODUCT", payload: 1}])
})



