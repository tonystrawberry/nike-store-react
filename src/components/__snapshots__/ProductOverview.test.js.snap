// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ProductOverview component 1`] = `
<Connect(ProductOverview)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProductOverview
    onSelected={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="product-overview"
      onClick={[Function]}
    >
      <div
        className="product-photo__container"
      >
        <img
          alt="Nike Air Force"
          className="product-photo"
          src="nike_airforce-small.jpg"
          srcSet="nike_airforce-small.jpg 216w"
        />
      </div>
      <div
        className="product-overview__info"
      >
        <h1
          className="product-overview__title"
        >
          Nike Air Force 1
        </h1>
        <div
          className="product-overview__type"
        >
          Men's Shoe
        </div>
        <div
          className="product-overview__type"
        >
          2 Colors
        </div>
      </div>
    </div>
  </ProductOverview>
</Connect(ProductOverview)>
`;
